openfisca_aotearoa/variables/acts/parental_leave/parental_leave.py

```diff
@@ -1,15 +1,106 @@
-
 # -*- coding: utf-8 -*-
 
 from openfisca_core.model_api import *
-from openfisca_aotearoa.entities import Person
+from openfisca_aotearoa.entities import Person, Family
 
 
 class parental_leave__is_primary_carer(Variable):
     value_type = bool
     entity = Person
     definition_period = MONTH
-    label = u"Is primary carer"
+    label = u"Is primary carer, as per Parental Leave and Employment Protection Act 1987"
+    reference = "http://www.legislation.govt.nz/act/public/1987/0129/latest/DLM120458.html"
+
+    def formula_2002(persons, period, parameters):
+        biological_mother = persons('parental_leave__is_the_biological_mother', period)
+
+        her_spouse = persons('parental_leave__is_spouse_or_partner_of_the_biological_mother', period)
+        received_transferred_entitlement = persons('parental_leave__has_spouse_who_transferred_her_entitlement', period)
+
+        other = persons('parental_leave__a_person_other_than_biological_mother_or_her_spouse', period)
+        permanent = persons('parental_leave__taking_permanent_primary_responsibility_for_child', period)
+
+        # Mark who is the principal caregiver, as there may be >1 eligible
+        # PPL Section 7 (2)
+        nominated = persons.has_role(Family.PRINCIPAL_CAREGIVER)
+
+        return nominated * (biological_mother + (her_spouse * received_transferred_entitlement) + (other * permanent))
+
+
+class parental_leave__is_the_biological_mother(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = ETERNITY
+    label = u"a female (the biological mother) who is pregnant or has given birth to a child"
+    reference = "http://www.legislation.govt.nz/act/public/1987/0129/latest/DLM120458.html"
+
+
+class parental_leave__is_spouse_or_partner_of_the_biological_mother(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = MONTH
+    label = u"the spouse or partner of the biological mother"
+    reference = "http://www.legislation.govt.nz/act/public/1987/0129/latest/DLM120458.html"
+
+    def formula_2002(persons, period, parameters):
+        # true for people who are not the biological mother, but the biological mother is in their family with role of partner
+        return persons.family('parental_leave__family_includes_biological_mother_as_partner', period) * \
+            not_(persons('parental_leave__is_the_biological_mother', period))
+
+
+class parental_leave__family_includes_biological_mother_as_partner(Variable):
+    value_type = bool
+    entity = Family
+    definition_period = MONTH
+    reference = "http://www.legislation.govt.nz/act/public/1964/0136/latest/whole.html#DLM361606"
```

Comment:
This seems to be an incorrect reference, it links to Orphans' benefits.

Reply:
WOAH! good catch

Reply:
FIXED